core/amber/src/main/scala/edu/uci/ics/amber/engine/architecture/scheduling/policies/AllReadyTimeInterleavedRegions.scala

```diff
@@ -0,0 +1,106 @@
+package edu.uci.ics.amber.engine.architecture.scheduling.policies
+
+import akka.actor.ActorContext
+import edu.uci.ics.amber.engine.architecture.controller.Workflow
+import edu.uci.ics.amber.engine.architecture.controller.promisehandlers.RegionsTimeSlotExpiredHandler.RegionsTimeSlotExpired
+import edu.uci.ics.amber.engine.architecture.scheduling.PipelinedRegion
+import edu.uci.ics.amber.engine.common.Constants
+import edu.uci.ics.amber.engine.common.amberexception.WorkflowRuntimeException
+import edu.uci.ics.amber.engine.common.rpc.AsyncRPCClient
+import edu.uci.ics.amber.engine.common.rpc.AsyncRPCClient.ControlInvocation
+import edu.uci.ics.amber.engine.common.virtualidentity.{ActorVirtualIdentity, LinkIdentity}
+
+import scala.collection.mutable
+import scala.concurrent.duration.{FiniteDuration, MILLISECONDS}
+import scala.jdk.CollectionConverters.asScalaSet
+import scala.util.control.Breaks.{break, breakable}
+
+class AllReadyTimeInterleavedRegions(
+    workflow: Workflow,
+    ctx: ActorContext,
+) extends SchedulingPolicy(workflow) {
+
+  var currentlyExecutingRegions = new mutable.LinkedHashSet[PipelinedRegion]()
+
+  override def checkRegionCompleted(region: PipelinedRegion): Unit = {
+    super.checkRegionCompleted(region)
+    if (isRegionCompleted(region)) {
+      currentlyExecutingRegions.remove(region)
+    }
+  }
+
+  override def onWorkerCompletion(workerId: ActorVirtualIdentity): Set[PipelinedRegion] = {
+    val region = getRegion(workerId)
```

Comment:
This function is a duplicate of the same function in the `SchedulingPolicy` base class, except for the last `if` statement. Is there any way to get rid of the duplicate? I'm wondering that why we don't need this check in the base class?

Reply:
discussed offline